app/javascript/components/settings-time-profile-form/index.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useState, useEffect } from 'react';
+import MiqFormRenderer from '@@ddf';
+import { Loading } from 'carbon-components-react';
+import PropTypes from 'prop-types';
+import createSchema from './settings-time-profile-form.schema';
+import { API } from '../../http_api';
+import miqRedirectBack from '../../helpers/miq-redirect-back';
+
+const SettingsTimeProfileForm = ({
+  timeProfileId, timezones, action, userid,
+}) => {
+  const [{
+    isLoading, initialValues, fields, tz,
+  }, setState] = useState({
+    isLoading: !!timeProfileId,
+    fields: [],
+  });
+
+  const [show, setShow] = useState(false);
+
+  const onCancel = () => {
+    let message = '';
+    if (timeProfileId !== '') {
+      message = __(`Editing of time profile ${initialValues.description} was cancelled by the user.`);
+    } else {
+      message = __(`Adding new time profile was cancelled by the user.`);
+    }
+    miqRedirectBack(message, 'warning', '/configuration/change_tab/4?uib-tab=4');
+  };
+
+  const onSubmit = (values) => {
+    if (values.profile.tz === 'null') {
+      values.profile.tz = null;
+    }
+
```

Comment:
can't we remove line no 36- 41, and can write this as below code?
`
if (values.profile.hoursAM. === undefined && values.profile.hoursPM ===undefined && values.HoursSelectAll === false) {
      setShow(true);
    } else {
      miqSparkleOn();`

Reply:
The reason I have the undefined and then the length 0 check is because if user does not touch the hours component it is initially undefined but if they check an hour then uncheck it it defined but as an empty array. I tried out your change but if I check an hour then uncheck it, the form lets me submit without error